specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/stable/2021-09-01/extensions.json

```diff
@@ -0,0 +1,864 @@
+          },
+          {
+            "$ref": "#/parameters/ExtensionNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "forceDelete",
+            "in": "query",
+            "description": "Delete the extension resource in Azure - not the normal asynchronous delete.",
+            "type": "boolean"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK"
+          },
+          "202": {
+            "description": "Accepted.  The request has been accepted for processing."
+          },
+          "204": {
+            "description": "No Content.  The request has been accepted but the extension was not found."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        }
+      },
+      "patch": {
+        "tags": [
+          "Extensions",
+          "ClusterExtensions"
+        ],
+        "description": "Patch an existing Kubernetes Cluster Extension.",
+        "operationId": "Extensions_Update",
+        "x-ms-examples": {
+          "Update Extension": {
+            "$ref": "./examples/PatchExtension.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../common/2021-09-01/definitions.json#/parameters/ClusterRpParameter"
+          },
+          {
+            "$ref": "../../common/2021-09-01/definitions.json#/parameters/ClusterResourceNameParameter"
+          },
+          {
+            "$ref": "../../common/2021-09-01/definitions.json#/parameters/ClusterNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ExtensionNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "patchExtension",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/patchExtension"
+            },
+            "description": "Properties to Patch in an existing Extension."
+          }
+        ],
+        "responses": {
+          "201": {
```

Comment:
>201 [](http://example.com/codeflow?start=11&length=3)

Long running patch should be using 202 as per [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#updating-using-patch](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#updating-using-patch)

Reply:
Yes, this one is interesting.  I too had it as 202 in my design; however, the Swagger tool complained that Patch must return 201 and I had to change it (in commit https://github.com/Azure/azure-rest-api-specs/pull/15593/commits/9150f46d8e6aa794d36bc210592b420288faf651)  Is there a way I can bypass this Swagger tool validation failure?  If so, I'd actually want to keep it as 202 per this api-reference recommendation.

Reply:
Sounds like a linter bug. @akning-ms can you advise here?

Reply:
I've changed it to 202, per guidance.